fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1552,7 +1609,19 @@ ACTOR Future<Void> reply(CommitBatchContext* self) {
 			if (r->value().size() && r->value().front().first < oldestVersion)
 				r->value().front().first = 0;
 		}
-		pProxyCommitData->keyResolvers.coalesce(allKeys);
+		if (SERVER_KNOBS->PROXY_USE_RESOLVER_PRIVATE_MUTATIONS) {
```

Comment:
What is this code block doing?

Reply:
2 things:

1. coalesce key ranges in `pProxyCommitData->keyResolvers`; and
2. reclaim older versions from `pProxyCommitData->systemKeyVersions`, similar to the logic in line 1605-1611 for `pProxyCommitData->keyResolvers`.